auth.go

```diff
@@ -162,6 +162,7 @@ func filterDefinition(sourceSchema *ast.Schema, visited map[string]bool, types m
 			if typ.Kind == ast.Interface {
 				for _, pt := range sourceSchema.PossibleTypes[typ.Name] {
 					types[pt.Name] = pt
+					_ = filterDefinition(sourceSchema, visited, types, pt, AllowedFields{AllowAll: true})
```

Comment:
how come the result is ok to throw away?

Reply:
The result of this function is a filtered definition for the sub type, but in that particular instance we are in the `AllowAll == true` branch, so we won't need that filtered definition as we're allowing everything.

Reply:
The result map is `types` so we add to this one, `visited` is just to make sure we don't create an infinite recursion.